src/compas_fab/robots/configuration.py

```diff
@@ -62,19 +92,53 @@ class Configuration(object):
 
     """
 
-    def __init__(self, values=None, types=None, joint_names=None):
+    def __init__(self, joint_values=None, types=None, joint_names=None):
+        joint_values = FixedLengthList(joint_values or [])
+        types = FixedLengthList(types or [])
+        joint_names = FixedLengthList(joint_names or [])
```

Comment:
It's currently possible to pass a list of `joint_names` with repeated names.
Here, Python is inconsistent is that, depending on the syntax you use, it will allow this or not, ie. `dict(a=1, a=3)` is a `SyntaxError`, but `{'a':1, 'a':3}` is last-one-wins. I find an error much better because it prevents a whole lot of other mistakes, so, I would check and raise an exception here.

Reply:
This is a very good point.

Reply:
is it ok to ignore the possibility that a user might do something like `config.joint_names[0] = config.joint_names[1] = 'a'`? otherwise modifying `FixedLengthList` to include optional validation in `__setitem__` and `__setslice__` would be really tricky.

Reply:
yep 👍